File: typescript/darts/darts.ts
export function score(x: number, y: number): number {
  const distanceToCenter = Math.sqrt(Math.pow(x, 2) + Math.pow(y, 2));

  if (distanceToCenter <= 1) {
    return 10;
  }
  if (distanceToCenter <= 5) {
    return 5;
  }
  if (distanceToCenter <= 10) {
    return 1;
  }
  return 0;
}

File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    distance_to_center = :math.sqrt(x * x + y * y)

    cond do
      distance_to_center <= 1 -> 10
      distance_to_center <= 5 -> 5
      distance_to_center <= 10 -> 1
      true -> 0
    end
  end
end
